coops-ui/tests/unit/OfficeAddresses.spec.ts

```diff
@@ -0,0 +1,411 @@
+    store.state.registeredAddress = {
+      deliveryAddress: {
+        addressCity: 'delCity',
+        addressCountry: 'delCountry',
+        addressRegion: 'delRegion',
+        deliveryInstructions: 'delInstructions',
+        postalCode: 'delPostal',
+        streetAddress: 'delStreet',
+        streetAddressAdditional: 'delStreetAdd'
+      },
+      mailingAddress: {
+        addressCity: 'mailCity',
+        addressCountry: 'mailCountry',
+        addressRegion: 'mailRegion',
+        deliveryInstructions: 'mailInstructions',
+        postalCode: 'mailPostal',
+        streetAddress: 'mailStreet',
+        streetAddressAdditional: 'mailStreetAdd'
+      }
+    }
+  })
+
+  const draftAddresses = {
+    registeredOffice: {
+      deliveryAddress: {
+        addressCity: 'delCityDraft',
+        addressCountry: 'delCountryDraft',
+        addressRegion: 'delRegionDraft',
+        deliveryInstructions: 'delInstructionsDraft',
+        postalCode: 'delPostalDraft',
+        streetAddress: 'delStreetDraft',
+        streetAddressAdditional: 'delStreetAddDraft'
+      },
+      mailingAddress: {
+        addressCity: 'mailCityDraft',
+        addressCountry: 'mailCountryDraft',
+        addressRegion: 'mailRegionDraft',
+        deliveryInstructions: 'mailInstructionsDraft',
+        postalCode: 'mailPostalDraft',
+        streetAddress: 'mailStreetDraft',
+        streetAddressAdditional: 'mailStreetAddDraft'
+      }
+    }
+  }
+
+  it('loads the current office addresses properly', async done => {
+    const Constructor = Vue.extend(OfficeAddresses)
+    const instance = new Constructor({
+      propsData: {
+        registeredAddress: store.state.registeredAddress,
+        recordsAddress: store.state.recordsAddress
+      },
+      store,
+      vuetify
+    })
+    vm = instance.$mount()
+
+    const deliveryAddress = vm.registeredAddress.deliveryAddress
+    expect(deliveryAddress['streetAddress']).toEqual('delStreet')
+    expect(deliveryAddress['streetAddressAdditional']).toEqual('delStreetAdd')
+    expect(deliveryAddress['addressCity']).toEqual('delCity')
+    expect(deliveryAddress['addressRegion']).toEqual('delRegion')
+    expect(deliveryAddress['postalCode']).toEqual('delPostal')
+    expect(deliveryAddress['addressCountry']).toEqual('delCountry')
+    expect(deliveryAddress['deliveryInstructions']).toEqual('delInstructions')
+
+    const mailingAddress = vm.registeredAddress.mailingAddress
+    expect(mailingAddress['streetAddress']).toEqual('mailStreet')
+    expect(mailingAddress['streetAddressAdditional']).toEqual('mailStreetAdd')
+    expect(mailingAddress['addressCity']).toEqual('mailCity')
+    expect(mailingAddress['addressRegion']).toEqual('mailRegion')
+    expect(mailingAddress['postalCode']).toEqual('mailPostal')
+    expect(mailingAddress['addressCountry']).toEqual('mailCountry')
+    expect(mailingAddress['deliveryInstructions']).toEqual('mailInstructions')
+
+    Vue.nextTick(() => {
```

Comment:
Is there a need to wait for the next tick here, or can you simply call done() ?

Actually, since you're not waiting for anything to complete before you tell this test it's done, you can remove done() and then just pass () into the arrow function on line 71.

Also below.

Reply:
100% right. I saw this implementation of nextTick and callbacks used consistently throughout the tests and just wanted to be congruent with the code as it was. But i whole heartedly agree, let's simplify as much as we can and remove it!

Reply:
Looks good. Keep the tests simple, yeah!